internal/directives/promotions.go

```diff
@@ -27,6 +27,18 @@ type PromotionStepRunner interface {
 	RunPromotionStep(context.Context, *PromotionStepContext) (PromotionStepResult, error)
 }
 
+// RetryableStepRunner is an interface for PromotionStepRunners that can be
+// retried in the event of a failure.
+type RetryableStepRunner interface {
```

Comment:
Super nit: `RetryableStepRunner` seems to imply that this interface is composed of `StepRunner` + additional methods. (In the same way an `io.ReadCloser` is _also_ an `io.Reader` and an `io.Closer`.)

I'm _not_ suggesting that you redefine it as such; rather I am wondering if `Retryable` might be a more apt name for the current definition.

I do not feel strongly about this at all, so completely disregard this if you think I'm bonkers. I may be.

Reply:
I didn't like _just_ `Retryable` because it seemed _too_ generic, which made me opt for this name instead.

Reply:
Ya. I can see that perspective as well and it's a big part of why I don't feel strongly about this. Let's just leave it be.